server/smtpd/lib.go

```diff
@@ -27,8 +27,8 @@ import (
 var (
 	// Debug `true` enables verbose logging.
 	Debug      = false
-	rcptToRE   = regexp.MustCompile(`[Tt][Oo]:\s?<(.+)>`)
-	mailFromRE = regexp.MustCompile(`[Ff][Rr][Oo][Mm]:\s?<(.*)>(\s(.*))?`) // Delivery Status Notifications are sent with "MAIL FROM:<>"
+	rcptToRE   = regexp.MustCompile(`[Tt][Oo]: ?<([^<>\v]+)>( |$)(.*)?`)
```

Comment:
The closing `( |$)(.*)?` should be just ` ?$` as we want to account for one possible space, and then the end of the line. Unlike `MAIL FROM`, I am not aware of any other arguments that are allowed to be passed to `RCPT TO` (unless you know something I don't?).

Reply:
`NOTIFY` and `ORCPT`, see https://datatracker.ietf.org/doc/html/rfc3461#section-4.1

Reply:
Good catch! OK, leave `rcptToRE` one as you had it :+1: